internal/http/services/owncloud/ocdav/propfind.go

```diff
@@ -275,6 +280,26 @@ func (s *svc) mdToPropResponse(ctx context.Context, pf *propfindXML, md *provide
 		Propstat: []propstatXML{},
 	}
 
+	role := conversions.RoleFromResourcePermissions(md.PermissionSet)
+
+	// files never have the create container permission
+	if md.Type == provider.ResourceType_RESOURCE_TYPE_FILE && md.PermissionSet != nil {
+		// we need to copy the permission set in case it was reused when passing it in while listing a collection
+		md.PermissionSet = copyPermissionSet(md.PermissionSet)
```

Comment:
Since we're assigning the permissions to the same variable, we do lose the original value. Why are we making a separate copy then?

Reply:
PermissionSet is a struct which is passed by reference. There are code paths that will change the PermissionSet  for files, because they never have delete or create permission IIRC. When listing a folder this would change the default PermissionSet and folders that are rendered after a file would retain the changed default.

Reply:
Yes but this particular function is called only from one place and we're forgetting the original reference (since we're storing the returned struct there) so essentially there's no way to get that back.

Reply:
ok, you are right ... I tried cleaning the permissions and roles stuff up. I think I was confusing this with the storage drivers ...

Reply:
all tests green!

Reply:
Perfect!